### PR TITLE
Feature: config siphon to take specific source registry yaml file

### PR DESCRIPTION
**Pull request by ShellyXueHan**
## Summary
Siphon is originally taking the source from registry.yml file.
we want to be able to config this, so that different registry.yml files would be picked in different environments.
This would allow the deployment in TEST environment to use fixtures and make test cases more stable.

### CODE REVIEW

**Comment by patricksimonian** [comment]
is this not ready for merge because it needs to be rebased?

**Comment by patricksimonian** [comment]
definitely needs to be rebased now.

**Comment by ShellyXueHan** [comment]
@patricksimonian did a rebase, will use `--interactive` next time to squash the commits.

File: app-web/gatsby-config.js [hunk 1]
```diff
@@ -70,6 +71,7 @@ module.exports = {
       options: {
         tokens: {
           GITHUB_API_TOKEN: process.env.GITHUB_TOKEN,
+          SOURCE_REGISTRY_TYPE: process.env.REGISTRY_TYPE || 'SourceRegistryYaml',
```
Comment (patricksimonian) [inline app-web/gatsby-config.js]: I think this should be moved out of tokens and be stand alone

Comment (patricksimonian) [reply app-web/gatsby-config.js]: also could we should perhaps document this somewhere just so we know how the source registry is acutally sourced. Its not too important now, but if this is ever exported as an npm package we will need explain the dependancy on gatsby source filesystem and gatsby transformer yaml plugins

**Pull request merged**